### PR TITLE
Azure-ARM builder: Deploy NSG if list of IP addresses is provided in config

### PR DESCRIPTION
**Pull request by sumit-kalra**
**Description of change**:
When Azure-ARM builder deploys a VM with a public IP; the VM is open to the entire internet with just a username-password auth. This PR is adding another, optional, level of security
- This change adds a config option to specify an array of IP addresses/CIDR blocks in the Azure-ARM builder
- If such an array is provided; Azure-ARM builder deploys a Network Security Group and associates it with the Virtual Network that it deploys.

**Testing**:
- Unit tests added. All unit tests: PASS
- Azure-ARM acceptance test: PASS
- Tested the feature (i.e. deploying an NSG): PASS

### CODE REVIEW

File: builder/azure/arm/config.go [hunk 1]
```diff
@@ -342,6 +343,13 @@ type Config struct {
 	// are None, ReadOnly, and ReadWrite. The default value is ReadWrite.
 	DiskCachingType string `mapstructure:"disk_caching_type" required:"false"`
 	diskCachingType compute.CachingTypes
+	// Specify the list of IP addresses and CIDR blocks that should be
+	// allowed access to the VM. If provided, an Azure Network Security
+	// Group will be created with corresponding rules and be bound to
+	// the NIC attached to the VM.
```
Comment (paulmey) [inline builder/azure/arm/config.go]: Looks like the NSG is bound to the subnet, not the NIC.

**Comment by SwampDragons** [comment]
We need you to sign the CLA before we can merge, but other than that this looks good.

**Comment by sumit-kalra** [comment]
> We need you to sign the CLA before we can merge, but other than that this looks good.

Thanks for looking @SwampDragons. I shall sign the CLA - waiting for some approvals at my end.

**Comment by sumit-kalra** [comment]
@paulmey @SwampDragons - Can you please merge this PR?

**Pull request merged**